### PR TITLE
Create `sdk` package

### PR DESCRIPTION
**Pull request by RedbackThomson**
Description of changes:
Creates a new `sdk` package that strictly serves the purpose of loading the operations and shapes from the AWS SDK. This is useful to isolate creating the custom fields that need to be injected during this stage.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: pkg/sdk/op.go [hunk 1]
```diff
@@ -11,7 +11,7 @@
 // express or implied. See the License for the specific language governing
 // permissions and limitations under the License.
 
-package model
+package sdk
```
[the remark this answers is not in the document]
Comment (RedbackThomson) [reply pkg/sdk/op.go]: I moved it into a new `operations` package, instead. `util` feels more like helper methods than functional logic.

File: pkg/generate/code/check.go [hunk 2]
```diff
@@ -65,21 +66,21 @@ func CheckExceptionMessage(
...
 	r *model.CRD,
-	opType model.OpType,
+	opType operations.OpType,
 	koVarName string,
 	indentLevel int,
 ) string {
 	var op *awssdkmodel.Operation
 	switch opType {
-	case model.OpTypeGet:
+	case operations.OpTypeGet:
```
Comment (A-Hilaly) [inline pkg/generate/code/check.go:75]: package `operations` should be renamed to `operation` and optionally we can rename `OpTypeGet` to `TypeGet` - https://rakyll.org/style-packages/

File: pkg/sdk/helper.go [hunk 3]
```diff
@@ -58,9 +66,9 @@ type SDKHelper struct {
 	APIGroupSuffix string
 }
 
-// NewSDKHelper returns a new SDKHelper object
-func NewSDKHelper(basePath string) *SDKHelper {
-	return &SDKHelper{
+// NewHelper returns a new SDKHelper object
+func NewHelper(basePath string) *Helper {
```
Comment (A-Hilaly) [inline pkg/sdk/helper.go:70]: :+1:

**Review by A-Hilaly — COMMENTED** [review]
looks good! Left one change request

**Comment by RedbackThomson** [comment]
As of this commit, I have merged `SDKAPI` back into the model package. I am using this type as an artifact produced by the `sdk.Helper`, which is now the only thing in the `sdk` package. Ideally the `SDKAPI` would be in there too, but it feels to tightly coupled to move at the moment.

File: pkg/model/types_test.go [hunk 4]
```diff
@@ -3,9 +3,8 @@ package model_test
 import (
 	"testing"
 
-	"github.com/stretchr/testify/assert"
-
 	"github.com/aws-controllers-k8s/code-generator/pkg/model"
+	"github.com/stretchr/testify/assert"
```
Comment (A-Hilaly) [inline pkg/model/types_test.go:7]: please undo

File: pkg/model/sdk_api.go [hunk 5]
```diff
@@ -14,177 +14,32 @@
...
-	"path/filepath"
-	"sort"
 	"strings"
 
-	"gopkg.in/src-d/go-git.v4"
-
 	ackgenconfig "github.com/aws-controllers-k8s/code-generator/pkg/generate/config"
 	"github.com/aws-controllers-k8s/code-generator/pkg/names"
 	"github.com/aws-controllers-k8s/code-generator/pkg/util"
-
```
Comment (A-Hilaly) [inline pkg/model/sdk_api.go:30]: ditto

File: pkg/generate/code/set_sdk.go [hunk 6]
```diff
@@ -1314,7 +1314,7 @@ func varEmptyConstructorK8sType(
 // setSDKForScalar returns the Go code that sets the value of a target variable
 // or field to a scalar value. For target variables that are structs, we output
 // the aws-sdk-go's common SetXXX() method. For everything else, we output
-// normal assignment operations.
+// normal assignment model.
```
Comment (jaypipes) [inline pkg/generate/code/set_sdk.go:1317]: think this is a find/replace oopsie.

File: pkg/generate/code/set_resource.go [hunk 7]
```diff
@@ -397,7 +397,7 @@ func setResourceReadMany(
 	// Find the element in the output shape that contains the list of
 	// resources. This heuristic is simplistic (just look for the field with a
 	// list type) but seems to be followed consistently by the aws-sdk-go for
-	// List operations.
+	// List model.
```
Comment (jaypipes) [inline pkg/generate/code/set_resource.go:400]: think this is a find/replace oopsie.

File: pkg/generate/code/set_resource.go [hunk 8]
```diff
@@ -307,7 +307,7 @@ func ListMemberNameInReadManyOutput(
...
```
Comment (jaypipes) [inline pkg/generate/code/set_resource.go:310]: think this is a find/replace oopsie.

File: pkg/model/sdk_api.go (hunk 5 above)
Comment (jaypipes) [reply pkg/model/sdk_api.go:30]: Generally we do:

```go
import (
   <std packages>

   <3rd party packages>

   <local packages>
)
```

so @A-Hilaly is saying the above should be:

```go
import (
	"strings"

	awssdkmodel "github.com/aws/aws-sdk-go/private/model/api"

	ackgenconfig "github.com/aws-controllers-k8s/code-generator/pkg/generate/config"
	"github.com/aws-controllers-k8s/code-generator/pkg/names"
	"github.com/aws-controllers-k8s/code-generator/pkg/util"
)
```

**Pull request merged**